v3/spring-cloud-aws-v3-autoconfigure/src/main/java/io/awspring/cloud/v3/autoconfigure/metrics/CloudWatchExportAutoConfiguration.java

```diff
@@ -0,0 +1,101 @@
+import io.awspring.cloud.v3.autoconfigure.core.RegionProviderAutoConfiguration;
+import io.awspring.cloud.v3.core.SpringCloudClientConfiguration;
+import io.micrometer.cloudwatch2.CloudWatchConfig;
+import io.micrometer.cloudwatch2.CloudWatchMeterRegistry;
+import io.micrometer.core.instrument.Clock;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.regions.providers.AwsRegionProvider;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClientBuilder;
+
+import org.springframework.boot.actuate.autoconfigure.metrics.CompositeMeterRegistryAutoConfiguration;
+import org.springframework.boot.actuate.autoconfigure.metrics.MetricsAutoConfiguration;
+import org.springframework.boot.actuate.autoconfigure.metrics.export.simple.SimpleMetricsExportAutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
+
+/**
+ * Configuration for exporting metrics to CloudWatch.
+ *
+ * @author Jon Schneider
+ * @author Dawid Kublik
+ * @author Jan Sauer
+ * @author Eddú Meléndez
+ * @since 2.0.0
+ */
+@Configuration(proxyBeanMethods = false)
+@AutoConfigureBefore({ CompositeMeterRegistryAutoConfiguration.class, SimpleMetricsExportAutoConfiguration.class })
+@AutoConfigureAfter({ CredentialsProviderAutoConfiguration.class, RegionProviderAutoConfiguration.class,
+		MetricsAutoConfiguration.class })
+@EnableConfigurationProperties(CloudWatchProperties.class)
+@ConditionalOnProperty(prefix = "management.metrics.export.cloudwatch", name = "namespace")
+@ConditionalOnClass({ CloudWatchAsyncClient.class, CloudWatchMeterRegistry.class, AwsRegionProvider.class })
+public class CloudWatchExportAutoConfiguration {
+
+	private final CloudWatchProperties properties;
+
+	public CloudWatchExportAutoConfiguration(CloudWatchProperties properties) {
+		this.properties = properties;
+	}
+
+	@Bean
+	@ConditionalOnProperty(value = "management.metrics.export.cloudwatch.enabled", matchIfMissing = true)
+	public CloudWatchMeterRegistry cloudWatchMeterRegistry(CloudWatchConfig config, Clock clock,
+			CloudWatchAsyncClient client) {
+		return new CloudWatchMeterRegistry(config, clock, client);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public CloudWatchAsyncClient amazonCloudWatchAsync(AwsCredentialsProvider credentialsProvider,
```

Comment:
perhaps bean name should be `cloudWatchClient`? or `cloudWatchAsyncClient`? We should avoid using "amazon", if we need it should be rather `AWS`, but definitely we need consistent naming for all client beans.

Reply:
agree!